File: db/src/main/java/com/delivery/db/repository/UserRepository.java
package com.delivery.db.repository;

import com.delivery.db.entity.User;
import com.delivery.db.enums.UserStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findFirstByIdAndStatusOrderByIdDesc(long id, UserStatus status);

    Optional<User> findFirstByEmailAndPasswordAndStatusOrderByIdDesc(String email, String password, UserStatus status);
}

File: db/src/main/kotlin/com/delivery/db/repository/UserRepository.kt
package com.delivery.db.repository

import com.delivery.db.entity.User
import com.delivery.db.enums.UserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun findFirstByIdAndStatusOrderByIdDesc(id: Long, status: UserStatus?): User?

    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): User?
}